fuse_core/CMakeLists.txt

```diff
@@ -8,22 +8,21 @@ set(build_depends
 find_package(catkin REQUIRED COMPONENTS
   ${build_depends}
 )
-find_package(Boost REQUIRED)
+find_package(Boost REQUIRED COMPONENTS serialization)
 find_package(Ceres REQUIRED)
 find_package(Eigen3 REQUIRED)
 
 catkin_package(
   INCLUDE_DIRS
     include
-    ${CERES_INCLUDE_DIRS}
-    ${EIGEN3_INCLUDE_DIRS}
   LIBRARIES
     ${PROJECT_NAME}
-    ${CERES_LIBRARIES}
   CATKIN_DEPENDS
     ${build_depends}
   DEPENDS
     Boost
+    CERES
+    EIGEN3
```

Comment:
Eigen is a header only library. I don't think there's any `EIGEN3_LIBRARIES` variable defined: https://bitbucket.org/eigen/eigen/src/default/cmake/Eigen3Config.cmake.in

Reply:
True. But the `DEPENDS` seems sufficiently flexible to handle one without the other:
https://github.com/ros/catkin/blob/kinetic-devel/cmake/catkin_package.cmake#L168

I did a quick test to see if I was crazy:
```
find_package(Eigen3 REQUIRED)
set(depend_name "EIGEN3")
set(STEVE_INCLUDE_DIRS "")
set(STEVE_LIBRARIES "")
list(APPEND STEVE_INCLUDE_DIRS ${${depend_name}_INCLUDE_DIRS})
list(APPEND STEVE_LIBRARIES ${${depend_name}_LIBRARIES})
message("STEVE_INCLUDE_DIRS: ${STEVE_INCLUDE_DIRS}")
message("STEVE_LIBRARIES: ${STEVE_LIBRARIES}")
```
And I get:
```
STEVE_INCLUDE_DIRS: /usr/include/eigen3
STEVE_LIBRARIES:
```
Which seems like the desired outcome.